scripts/upstreamMerge.sh

```diff
@@ -37,7 +37,7 @@ function main {
     cd "${basedir}"
     ./gradlew cleanCache || exit 1 # todo: Figure out why this is necessary
     ./gradlew applyPatches -Dpaperweight.debug=true || exit 1
-    ./gradlew rebuildPatches || exit 1
+    ./gradlew rebuildPatches --filter-patches=false || exit 1
```

Comment:
```
See "git help gc" for manual housekeeping.
See "git help gc" for manual housekeeping.
   Patches applied cleanly to Paper-Server

BUILD SUCCESSFUL in 9m 34s
36 actionable tasks: 31 executed, 4 from cache, 1 up-to-date

FAILURE: Build failed with an exception.

* What went wrong:
Problem configuring task :rebuildPatches from command line.
> Unknown command-line option '--filter-patches'.

* Try:
Run gradlew help --task :rebuildPatches to get task usage details. Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1s
```
apparently this is not a thing?

Reply:
cool cool

![https://i.imgur.com/P903yvW.png](https://i.imgur.com/P903yvW.png)

https://github.com/PaperMC/paperweight/blob/f5d09e8d1d162a8371dcae79791abe3c12ff3013/paperweight-lib/src/main/kotlin/tasks/RebuildPaperPatches.kt#L52

gradle moment?

Reply:
ok, turns out you can't pass that option to the rebuildPatches task, you need to pass it to the rebuildApiPatches and rebuildServerPatches tasks, this is a documented limitation of gradle options https://docs.gradle.org/current/userguide/custom_tasks.html#limitations

Reply:
I have no idea how I tested it to not catch it. Also, apparently, you can't disable any gradle command-line options through CLI so this doesn't work at all: https://github.com/gradle/gradle/issues/4311

Reply:
Reverted this for now, as this was additional thing I did and don't want it to hold other PR and internal stuff blocked.

Reply:
Updated to use the property introduced in latest paperweight: https://github.com/PaperMC/paperweight/commit/1e3d63a29b659cde2ca8ee4944cbfc888b821541
It does nothing for now, but will automagically work when Paper updates the paperweight version it uses.
Also, at @jpenilla request, the script will rebuild patches without filtering before updating submodules, then do it again once submodules are updated. If submodules introduce no changes, changes to all patches will be reverted.

I could probably also make it first check, if any of the submodules requested for update have changes (do `git fetch` and `git diff --exit-code --quiet origin/master`), and only if changes are expected, do the whole process?